### ISSUE TITLE
Add Support for Separate Pull Requests for Major, Minor, and Patch Version Updates

### ISSUE THREAD

**Issue by ClemensRau1337** [dependabot/dependabot-core#6957]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature description

Currently, Dependabot does not provide a native way to separate Pull Requests for major, minor, and patch version updates within the same configuration. As a result, users cannot easily distinguish between different update types and may unintentionally merge major version updates that require more extensive testing and refactoring.

**Feature Request:**

I propose the addition of a feature that allows Dependabot to create separate Pull Requests for major, minor, and patch version updates. This would enable users to better manage updates by having dedicated Pull Requests for each update type, making it easier to prioritize and review changes based on their potential impact.

**Example:**

```
version: 2
updates:
  - package-ecosystem: "npm"
    directory: "/"
    schedule:
      interval: "weekly"
    commit-message:
      prefix: "Dependabot Minor/Patch"
    versioning-strategy: "increase"
    update-types: ["version-update:semver-minor", "version-update:semver-patch"]

  - package-ecosystem: "npm"
    directory: "/"
    schedule:
      interval: "weekly"
    commit-message:
      prefix: "Dependabot Major"
    versioning-strategy: "increase"
    update-types: ["version-update:semver-major"]
```

**Comment by jeffwidman** [comment]
I think this is effectively a duplicate of:
* https://github.com/dependabot/dependabot-core/issues/1778

Your example is a slightly different example of it, so I've copied the description over for us to consider if we ever build out that feature. 

Please 👍 that ticket if it's something you want.

**Comment by yeikel** [comment]
> There is value to doing incremental bumps rather than always jumping straight to latest, but from an implementation perspective that's difficult based on our current architecture, and frankly I'm not clear on the value... 

A common scenario that could shed light here is when a library releases a new breaking version, v2, along with a non-breaking incremental patch for v1.1 to address security issues. It's common to upgrade to v1.1 as an interim solution while still keeping the pull request for v2 open for further planning and capacity allocation.

An instance of this scenario can be observed in graphql-java, where the breaking v20 release was made while maintaining v19.x simultaneously, with releases like v19.4 and then v19.5. In this example, my team is not ready to upgrade to v20 yet but we monitored the project manually and upgraded to 19.4 and 19.5 manually. It would be better if dependabot could detect this

**Comment by brcarp** [comment]
> In this example, my team is not ready to upgrade to v20, but we manually monitored the project and upgraded to 19.4 and 19.5. It would be better if dependabot could detect this

Dependabot _does_ have this in the sense that you can instruct Dependabot to ignore a major version (in its PR that upgrades to the next major version), and then it will stop doing that while continuing to produce minor/patch updates.

**Comment by yeikel** [comment]
> > In this example, my team is not ready to upgrade to v20, but we manually monitored the project and upgraded to 19.4 and 19.5. It would be better if dependabot could detect this
> 
> Dependabot _does_ have this in the sense that you can instruct Dependabot to ignore a major version (in its PR that upgrades to the next major version), and then it will stop doing that while continuing to produce minor/patch updates.

We do not want to ignore the major version. In fact, it would be ideal to continue hearing about new releases of the major version as well

**Comment by Aaron-Ritter** [comment]
We have a very similar scenario we are currently testing:

```
  # Maintain dependencies for npm
  - package-ecosystem: "npm"
    directory: "/"
    schedule:
      interval: "daily"
    groups:
      prod-npm-minor-dependencies:
        dependency-type: "production"
        update-types:
        - "minor"
        - "patch"
      dev-npm-minor-dependencies:
        dependency-type: "development"
        update-types:
        - "minor"
        - "patch"
      prod-npm-major-dependencies:
        dependency-type: "production"
        update-types:
        - "major"
      dev-npm-major-dependencies:
        dependency-type: "development"
        update-types:
        - "major"
```

Reason behind this, all thought it might look strange in the beginning, we can clearly differentiate the relevance of dev and prod related packages from a build process perspective, and as mentioned earlier we want to know the major changes and the minor/patch changes separately as major usually require manual work and minor could be merged without any manual intervention.

**Comment by asbjornu** [comment]
@jeffwidman,

> There is value to doing incremental bumps rather than always jumping straight to latest, but from an implementation perspective that's difficult based on our current architecture

As @brcarp writes in https://github.com/dependabot/dependabot-core/issues/6957#issuecomment-1502043991, Dependabot seems to already have the architecture necessary to support this by being able to differentiate between `update-types: [major, minor, patch]`.

> and frankly I'm not clear on the value... if you have a particular version you want to update to, then why not do it manually?

With many dependencies, especially across different package ecosystems, it's hard to keep up with all releases. Being notified of a major release gives us time to plan for it and prioritize the changes required internally to accommodate the breaking changes in the dependency. 

Separating major releases from minor ones gives us the ability to maintain the product while planning and working on the major version upgrade is happening in parallel on a separate branch.

**Comment by SamJUK** [comment]
+1 Coming from the PHP/Magento ecosystem, it is very common to see packages receive ongoing support for multiple Major versions. 

Minor/Patch updates we typically apply straight away via CI as the risk is low. 
Major versions usually take a while to approve due to breaking changes, infra updates, and additional testing. 

Whilst the supporting work is underway to implement the latest Major version, we still want to make sure we are keeping up on updates for the current Major version.

Real World Example, the [Adyen Module](https://github.com/Adyen/adyen-magento2/tags) currently supports both v9.x.x and v8.x.x versions.

**Comment by yeikel** [comment]
I placed https://github.com/dependabot/dependabot-core/issues/13665 as a feature request for Groups specifically as it may be doable to do this in the context of groups 👀
